### DISCUSSION TITLE
Mortality estimation when using censored values above highest event value in leaf

### DISCUSSION THREAD

**Discussion by derrynknife** [kogalur/randomForestSRC#375] [Q&A]
When estimating the ensemble mortality using (page 846 from [here](https://ishwaran.org/papers/IKBL.AOAS.pdf)):

![Screenshot 2023-04-23 at 12 31 26 pm](https://user-images.githubusercontent.com/18457108/233816347-f3d39005-0a15-4f0e-83c4-b264cecdf4b8.png)

How does the mortality estimate work when Tj is above the highest event value from the leaf? Do we just sum the CHF over the range of failures or is something else needed?

**Comment by ishwaran** [comment]
The package calculates mortality by summing the OOB CHF over the event times.  Here's a snippet of code to make this more clear:


```
data(pbc, package = "randomForestSRC")
o <- rfsrc(Surv(days, status) ~ ., pbc)

## manually calculate mortality
mort <- rowSums(o$chf.oob)

## difference between manual and direct calculation
> print(summary(mort - o$predicted.oob))

      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-7.105e-14 -3.553e-15  8.882e-16  5.773e-15  7.105e-15  1.279e-13 

```

The steps for calculating mortality for a case are as follows.  Consider only the trees where the case is OOB.  Over each of these trees identify the leaf (terminal node) and extract the Nelson-Aalen (NA) estimator for the case.  This is a step function taking values at each of the event times (event times defined by the full data, not the tree data).  Average the NA estimator of the OOB trees.  This gives the OOB CHF for the case.  The mortality for the case equals the summed value of its OOB CHF.

**Reply by derrynknife** [reply]
Thanks, this is a great explanation, much appreciated.

**Comment by derrynknife** [comment]
Thanks but let's say a leaf used the times t <- c(1, 2, 3, 4, 5) and the OOB event time is 6. How do we calculate the sum of the CHF. If the CHF is CHF <- c(1/4, 9/20, 47/60, 77/60, 137/60) for each time in t (using the NA estimator). How do we account for time 6?

Do we do a cum of CHF as if there is only 5 values or do we add another 137/60 assuming the CHF "plateaus" after the final event time. i.e is it this:

M = sum(c(1/4, 9/20, 47/60, 77/60, 137/60))

OR:

M = sum(c(1/4, 9/20, 47/60, 77/60, 137/60, 137/60))

???

**Comment by ishwaran** [comment]
It doesn't matter what times are in the leaf.  The NA estimator is calculated at each time point from a master list of time values: this master list of values is defined to be the event times of the full (entire) data set.  It's true that the NA estimator for the leaf will only take jumps at the observed event times in the leaf, but it's still a function defined over all time event values.  We sum the NA over all the event times.
